probe/appclient/report_publisher.go

```diff
@@ -8,18 +8,37 @@ import (
 // A ReportPublisher uses a buffer pool to serialise reports, which it
 // then passes to a publisher
 type ReportPublisher struct {
-	publisher Publisher
+	publisher  Publisher
+	noControls bool
 }
 
 // NewReportPublisher creates a new report publisher
-func NewReportPublisher(publisher Publisher) *ReportPublisher {
+func NewReportPublisher(publisher Publisher, noControls bool) *ReportPublisher {
 	return &ReportPublisher{
-		publisher: publisher,
+		publisher:  publisher,
+		noControls: noControls,
 	}
 }
 
+func removeControls(r report.Report) report.Report {
```

Comment:
My initial thought was to put this logic in the individual probe modules that populate the controls (ie so nodes don't get controls to begin with), but this is much cleaner!

Is there a way to do this without maintaining the list of topologies here?  We have some helper functions for iterating over topologies in report.go.